app/concerns/long_runnable.rb

```diff
@@ -65,6 +66,7 @@ def run
 
     def run!
       @thread = Thread.new do
+        Thread.current[:name] = "#{id}-#{Time.now}"
```

Comment:
It should fix two issues, first first is a possible race condition: When the `restart_dead_workers` check ran between the time where the worker was [stopped](https://github.com/dsander/huginn/blob/fix-twitter-agent-restart-leak/app/concerns/long_runnable.rb#L106) two lines later [restarted](https://github.com/dsander/huginn/blob/fix-twitter-agent-restart-leak/app/concerns/long_runnable.rb#L108) it would assume the worker crashed because the thread is dead and start another instance of it.
The second problem was that `TwitterStreamAgent::Worker` [stops](https://github.com/dsander/huginn/blob/fix-twitter-agent-restart-leak/app/models/agents/twitter_stream_agent.rb#L174) it's thread and it seems like that sleeping threads are not terminating them self, this is fixed by the added `wakeup` call.